static/js/panel.js

```diff
@@ -126,4 +172,69 @@ var Panel = new (class Panel {
     this.content.setAttribute("aria-expanded", !hidden);
     this.icon.classList.toggle("expanded");
   }
+
+  copyText(copy, text) {
+    copy.setAttribute("data-copying", "true");
+    navigator.clipboard.writeText(text)
+      .then(function() {
+        copy.classList.add("copied");
+        setTimeout(function() {
+          if (!copy.hasAttribute("data-copying")) {
+            copy.classList.remove("copied");
+          }
+        }, 1000);
+      })
+      .finally(function() {
+        copy.removeAttribute("data-copying");
+      });
+  }
+
+  copyMarkdown(type) {
+    const { node, getText } = this.markdown[type];
+    if (node.disabled) {
+      return;
+    }
+
+    const copy = node.querySelector(".copy");
+    const url = this.permalinkNode ? this.permalinkNode.href : document.location.href;
+    const text = getText(url);;
+
+    this.copyText(copy, text);
+  }
+
+  formatSelectedLines() {
+    const texts = [];
+    let lastLine = -1;
+    for (const line of [...Highlight.selectedLines].sort((a, b) => a - b)) {
+      if (lastLine !== -1 && lastLine != line - 1) {
+        texts.push("...");
+      }
+
+      const lineElem = document.getElementById(`line-${line}`).querySelector(".source-line");
+      texts.push(lineElem.textContent.replace(/\n/g, ""));
+
+      lastLine = line;
+    }
+    return texts;
+  }
+
+  updateMarkdownState() {
+    for (const [_, { node, isEnabled }] of Object.entries(this.markdown)) {
+      if (isEnabled()) {
+        node.disabled = false;
+        node.removeAttribute("aria-disabled");
```

Comment:
This could be written:

```
node.disabled = !isEnabled();
node.toggleAttribute("aria-disabled", node.disabled);
```

But this looks good as well.

Reply:
The ARIA spec seem to expect the value being "true", "false", or unset
https://w3c.github.io/aria/#enumeration-example

`toggleAttribute` sets it to empty string, so I guess `setAttribute`+`removeAttribute` is safer.